Assets/Scripts/SceneControllers/GameSceneController.cs

```diff
@@ -102,23 +102,30 @@ public void OnGameOver()
 
   public void OnRestartGame()
   {
-    AudioManager.instance.PlaySound("ClickSFX", false);
-    CountdownIsOn = true;
-    StartCoroutine(countdownUI.GetComponent<CountdownController>().StartCountdown());
-    playerController.OnRestartGame();
-    LevelGenerator levelGenerator = gameObject.GetComponent<LevelGenerator>();
-    levelGenerator.RestartGame();
-
-    AudioManager audioManager = AudioManager.instance;
-    audioManager.PlaySound("BackgroundMusic", true);
-
-    parallaxBackground.OnRestartGame();
-
-    gameUI.gameObject.SetActive(true);
-    gameOverUI.gameObject.SetActive(false);
-
+    
+    Scene scene = SceneManager.GetActiveScene(); 
     GameIsOver = false;
     Time.timeScale = 1;
+    SceneManager.LoadScene(scene.name);
+
+    
+    // AudioManager.instance.PlaySound("ClickSFX", false);
```

Comment:
Please, don't remove `AudioManager.instance.PlaySound("ClickSFX", false);`, we need that line for emitting a sound when clicking restart button.

Can you also remove all these commented lines?